packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -90,6 +93,24 @@ class ImageCache {
   int get currentSizeBytes => _currentSizeBytes;
   int _currentSizeBytes = 0;
 
+  /// Callback that is executed when inserting an image whose byte size is
+  /// larger than the [maximumByteSize].  Editing the [maximumByteSize] in the
+  /// callback can accomodate for the image.  Set to `null` for the default
+  /// behavior, which is to increase the [maximumByteSize] to accomodate the
+  /// large image.
+  ///
+  /// Here is an example implementation that increases the cache size in
+  /// response to a large image:
+  /// static void handler(ImageCache imageCache, int imageSize) {
+  ///   final int newSize = imageSize + 1000;
+  ///   imageCache.maximumSizeBytes = newSize;
+  ///   Analytics.log("Increase image cache size", newSize);
+  /// }
```

Comment:
If you grep the repo for `{@tool sample` you can see how we format code samples.  I think this would be better as a little bit more of a sample (and at the very least formatted with markdown for the code).

Reply:
Done

Reply:
Analyzer isn't happy with this

Reply:
Oh, I see now haha

Reply:
Nit: can you fill out the sample a little bit more, e.g. showing how to call it from main and when to call it?

Reply:
I would like to avoid documenting things outside of the scope of the class in docstrings.  That helps people focus on where to find information and keeps docstrings up to date.

I went to add a "See also" section that tells where to get the singleton but we already have one on the class docstring.  I don't think it makes sense to put a "See also:" in front of every public method.